irobotclient/test/test_main.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+from unittest.mock import MagicMock
+
+import requests
+import os
+
+import irobotclient.main
+
+
+class TestMain(unittest.TestCase):
+
+    test_data_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdata/')
+    output_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdir/')
+
+    def setUp(self):
+        self._old_request_head = requests.head
+        self._response = requests.Response()
+        requests.head = MagicMock(spec=requests.head)
+        requests.head.return_value = self._response
+
+    def tearDown(self):
+        requests.head = self._old_request_head
+
+    def test_download_binary(self):
+        test_file = "test.cram"
+        self._response.status_code = 200
+        self._response.url = self.test_data_dir + test_file
+        self._response.headers["content-type"] = "application/octet-stream"
```

Comment:
I've removed this piece of code now but I'll remember it for the future and check my current implementation for magic strings that can be... constant-ised :-)